compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -485,19 +485,21 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    TR_ASSERT(hasCompPtrs, "no compression sequence found under l2a node [%p]\n", node);
    */
 
-   // TODO (GuardedStorage)
-   TR::Register *source;
-   if (comp->useCompressedPointers() && (TR::Compiler->om.compressedReferenceShift() == 0 || firstChild->containsCompressionSequence()) && !node->isl2aForCompressedArrayletLeafLoad())
-      cg->setEvalCompressionSequence(true);
+   TR::Register* source = NULL;
 
-   if ((TR::Compiler->target.cpu.getS390SupportsGuardedStorageFacility()) && comp->useCompressedPointers() && (firstChild->containsCompressionSequence()) && !node->isl2aForCompressedArrayletLeafLoad())
+   // TODO (GuardedStorage): Why is isl2aForCompressedArrayletLeafLoad check necessary?
```

Comment:
This is a bug. I was under the impression that `containsCompressionSequence` would return true for `compressedReferenceShift() == 0` and hence I thought the check was redundant. To my surprise this is not true and the check is needed. Fixed in a4bdb5b.

As an aside I'm curious if anyone can answer why a `compressedReferenceShift() == 0` tree is not defined as a compression sequence? It technically is a compression sequence with a shift of zero.